src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,186 @@
+          // Starts a new transaction (acquiring and updating of items performs in single transaction)
+          conn.begin()
+              .compose(tx -> conn
+
+                  // Acquiring current bunch of items
+                .preparedQuery(SQL_SELECT_ITEMS)
+                .execute(Tuple.of(updateBunchSize))
+                .compose(itemsRows -> {
+                  final Promise<RowSet<Row>> promiseOfUpdateItems = Promise.promise();
+
+                  // Read items and fill them into data structures for further processing
+                  List<Item> itemList = new ArrayList<>();
+                  itemsRows.forEach(row -> {
+                    UUID itemId = row.getUUID("id");
+                    JsonObject itemJsonObject = row.getJsonObject("itemJson");
+                    Item item = itemJsonObject.mapTo(Item.class);
+                    itemList.add(item);
+                    itemsMap.put(itemId, item);
+                  });
+
+                  // Calculate items shelving order
+                  Map<UUID, JsonObject> updatedItemsMap = itemList.stream()
+                      .map(EffectiveCallNumberComponentsUtil::getCalculateAndSetEffectiveShelvingOrder)
+                      .collect(Collectors.toMap(item -> UUID.fromString(item.getId()), JsonObject::mapFrom));
+
+                  // Prepare items update parameters
+                  List<Tuple> itemsUpdateParams = new ArrayList<>();
+                  updatedItemsMap.entrySet().stream()
+                      .map(entry -> itemsUpdateParams.add(Tuple.of(entry.getValue(), entry.getKey())))
+                      .collect(Collectors.toList());
+
+                  // Updates of items
+                  conn
+                      .preparedQuery(SQL_UPDATE_ITEMS)
+                      .executeBatch(itemsUpdateParams, res -> {
+                        if (res.succeeded()) {
+                          int updatedItemsCount = res.result().iterator().next().getInteger(0);
+                          log.info("There were {} items updated", updatedItemsCount);
+                          promiseOfUpdateItems.complete(res.result());
+                        } else {
+                          log.error("Items updates failed {}", res.cause());
+                          promiseOfUpdateItems.fail(res.cause().getMessage());
+                        }
+                      });
+
+                  return promiseOfUpdateItems.future();
+                }
+              )
+                  // Finish transaction
+              .compose(res3 -> tx.commit())
+              .eventually(v -> conn.close())
+                  .onSuccess(h -> log.info("Items update transaction succeeded"))
+                  .onFailure(h -> log.error("Items update transaction filed: {}", h.getMessage())));
+
+        } // Connection acquired successfully
+
+      }); // Eof getConnection scope
+
+     }); // Eof vert.x timer scope
+
+    return promise.future();
```

Comment:
When does this promise complete? I couldn't find any code that completes this promise.

Reply:
Good point, seems missed setting it up  when current transaction finished.